### PR TITLE
API integration for email/password factor

### PR DESCRIPTION
**Pull request by nikhilvenkatraman-okta**
Resolves: OKTA-245296,OKTA-245295

### CODE REVIEW

File: src/v2/view-builder/internals/FormInputFactory.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const changeLabelToTop = (opt) => {
 const createFactorTypeView = (opt) => {
   var optionItems = (opt.options || [])
     .map(opt => {
-      return Object.assign({}, opt, FactorUtil.getFactorData(opt.value));
+      return Object.assign({}, opt, FactorUtil.getFactorData(opt.factorType));
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/internals/FormInputFactory.js]: This is dependent on https://oktainc.atlassian.net/browse/OKTA-248700

File: src/v2/ion/uiSchemaTransformer.js [hunk 2]
```diff
@@ -56,7 +56,10 @@ const createUISchema = (remediationValue = []) => {
       };
     }
     //select factor form for multiple factor enroll and multiple factors verify
-    if (ionFormField.name === 'factorType' || ionFormField.name === 'factorId') {
+    // when factor has not been enrolled we get back factorProfileId form the API
+    if (ionFormField.name === 'factorType' ||
+      ionFormField.name === 'factorId' ||
+      ionFormField.name === 'factorProfileId') {
```
Comment (haishengwu-okta) [inline src/v2/ion/uiSchemaTransformer.js]: are all three still relevant or can choose specific ones?

File: src/v2/view-builder/views/email/OTPEmailView.js [hunk 3]
```diff
@@ -48,21 +48,30 @@ const Body = BaseForm.extend(Object.assign(
...
+    'An email has been sent to you. Please click the link in your email or enter the code from that email below.',
 
     saveForm () {
       BaseForm.prototype.saveForm.apply(this, arguments);
       this.stopPolling();
       // TODO: abort ongoing request. (https://oktainc.atlassian.net/browse/OKTA-244134)
     },
 
+    sendEmailLink () {
+      _.delay(_.bind(function () {
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/email/OTPEmailView.js]: add comment why we do this and how you choose `300ms` (guessing randomly?)

File: src/v2/view-builder/ViewFactory.js [hunk 4]
```diff
@@ -41,7 +40,7 @@ const VIEWS_MAPPING = {
     password: EnrollFactorPasswordView
   },
   'challenge-factor': {
-    email: RequiredFactorEmailView,
+    email: OTPEmailView,
```
Comment (haishengwu-okta) [inline src/v2/view-builder/ViewFactory.js]: would rather delete `OTPEmailView` and merge code into `RequiredFactorEmailView.js`

File: src/v2/view-builder/views/email/OTPEmailView.js (hunk 3 above)
Comment (nikhilvenkatraman-okta) [reply src/v2/view-builder/views/email/OTPEmailView.js]: Yeah, no particular reason. Just waiting for the form to finishing rendering and start polling.

File: src/v2/view-builder/ViewFactory.js (hunk 4 above)
Comment (nikhilvenkatraman-okta) [reply src/v2/view-builder/ViewFactory.js]: Will rename OTPEmailView as RequiredFactorEmailView

File: src/v2/ion/uiSchemaTransformer.js (hunk 2 above)
Comment (nikhilvenkatraman-okta) [reply src/v2/ion/uiSchemaTransformer.js]: Like my comment describes we'll need `factorId ` and `factorProfileId `. Waiting to confirm with API folks to see if we can remove check on `factorType `.

**Pull request merged**